### ISSUE TITLE
Prettier html classes multiline not affected by printWidth config 

### ISSUE THREAD

**Issue by gkatsanos** [prettier/prettier#12048]
I have a different behavior of Prettier between the VS Code Extension and the CLI.
**prettier versions installed**
```sh
$ yarn why prettier
yarn why v1.22.17
[1/4] 🤔  Why do we have the module "prettier"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "prettier@2.5.1"
info Has been hoisted to "prettier"
info This module exists because it's specified in "devDependencies".
info Disk size without dependencies: "20.12MB"
info Disk size with unique dependencies: "20.12MB"
info Disk size with transitive dependencies: "20.12MB"
info Number of shared dependencies: 0
=> Found "@vue/component-compiler-utils#prettier@1.19.1"
info This module exists because "nuxt#@nuxt#webpack#vue-loader#@vue#component-compiler-utils" depends on it.
info Disk size without dependencies: "9.16MB"
info Disk size with unique dependencies: "9.16MB"
info Disk size with transitive dependencies: "9.16MB"
info Number of shared dependencies: 0
✨  Done in 0.34s.
```
**configuration:**
```json
{
  "semi": false,
  "singleQuote": true,
  "printWidth": 120
}
```

**input:**
```html
<template>
  <vis-card>
    <template #title>
      {{ $t('product_details.catalogue.title') }}
    </template>
    <a
      class="flex items-center border rounded mb-1 p-2 border-navy-20 text-blue-100 hover:border-blue-120 hover:text-blue-120 cursor-pointer"
      :href="$device.isMobileOrTablet ? catalogueUrl : cataloguePageUrl"
      target="_blank"
      data-test="catalogue"
      @click="handleClickOnCatalogue"
    >
      <vis-icon class="" width="24" height="24">
        <icon-pdf />
      </vis-icon>
      <span class="ml-1 mr-auto">{{ $t('product_details.catalogue.button') }}</span>
      <vis-icon class="text-navy-100" width="24" height="24">
        <icon-eye />
      </vis-icon>
    </a>
  </vis-card>
</template>
```

I expect `printWidth: 120` to format the line below into multilines :
```
      class="flex items-center border rounded mb-1 p-2 border-navy-20 text-blue-100 hover:border-blue-120 hover:text-blue-120 cursor-pointer"
```

this seems to be the case when using the VsCode extension:

```html
    <a
      class="
        flex
        items-center
        border
        rounded
        mb-1
        p-2
        border-navy-20
        text-blue-100
        hover:border-blue-120 hover:text-blue-120
        cursor-pointer
      "
      :href="$device.isMobileOrTablet ? catalogueUrl : cataloguePageUrl"
      target="_blank"
      data-test="catalogue"
      @click="handleClickOnCatalogue"
    >
```

But when running Prettier via the CLI, it doesn't multiline:

```sh
$ yarn run prettier components/product-details/secondary-column/Catalogue.vue --config .prettierrc --write --loglevel=debug
yarn run v1.22.17
$ /Users/xxxxxx/repos/product-details-frontend/node_modules/.bin/prettier components/product-details/secondary-column/Catalogue.vue --config .prettierrc --write --loglevel=debug
[debug] normalized argv: {"_":["components/product-details/secondary-column/Catalogue.vue"],"color":true,"editorconfig":true,"write":true,"config":".prettierrc","loglevel":"debug","config-precedence":"cli-override","debug-repeat":0,"ignore-path":".prettierignore","plugin":[],"plugin-search-dir":[]}
[debug] load config file from '.prettierrc'
[debug] loaded options `{"semi":false,"singleQuote":true,"printWidth":120}`
[debug] applied config-precedence (cli-override): {"printWidth":120,"semi":false,"singleQuote":true}
components/product-details/secondary-column/Catalogue.vue 44ms
✨  Done in 0.89s.
```

**output**
```html
    <a
      class="flex items-center border rounded mb-1 p-2 border-navy-20 text-blue-100 hover:border-blue-120 hover:text-blue-120 cursor-pointer"
      :href="$device.isMobileOrTablet ? catalogueUrl : cataloguePageUrl"
      target="_blank"
      data-test="catalogue"
      @click="handleClickOnCatalogue"
    >
```

Which is the correct output and why? I assume there's some config mismatch but I just would like to understand if multiline is related to printWidth or not.  
(subsequently how come the Prettier VSCode formatter is formatting it correctly but I understand this might be a question for their issue tracker!)

**Comment by Balastrong** [comment]
Hello!
I see your case involves a string (the value of class attribute) that is not properly formatted, I'd initially assume because it's a string
However, I've also got a similar case where the same happens (prettier from cli does not break the line, while VSCode extension does) on plain code.

Specifically, this line:

```ts
  public selectedSpellInDriverSchedule: SpellInDriverScheduleDetailViewModelDTO = new SpellInDriverScheduleDetailViewModelDTO();
```

Is kept as is (longer than 120) upon formatting via cli, but is broken on formatting via VSCode extension, as follows:

```ts
  public selectedSpellInDriverSchedule: SpellInDriverScheduleDetailViewModelDTO =
new SpellInDriverScheduleDetailViewModelDTO();
```

This is really annoying on my project as we have a git hook to format on commit, hence properly formatted code constantly gets edited on commits.

**Comment by duailibe** [comment]
@gkatsanos @Balastrong running Prettier in the CLI is most likely the correct output. If formatting via VSCode gives a different output, there's probably another formatter. In HTML, maybe it was [this](https://code.visualstudio.com/docs/languages/html)?

You can also refer to https://prettier.io/playground to check how Prettier formats things given the various languages and settings.

**Comment by Balastrong** [comment]
I'm 100% sure the selected formatter on VSCode is Prettier, it also prints in Prettier's console when the document is formatted via VSCode shortcut.

This is the console output on VSCode:

```
["INFO" - 18:19:15] Extension Name: esbenp.prettier-vscode.
["INFO" - 18:19:15] Extension Version: 9.2.0.
["INFO" - 18:19:27] Formatting file:///e%3A/PlusProgetti/Net.Pluservice.Motus/Unbounded/src/app/views/driverScheduleVersion/driverScheduleVersion.component.ts
["INFO" - 18:19:27] Using config file at 'e:\PlusProgetti\Net.Pluservice.Motus\Unbounded\.prettierrc'
["INFO" - 18:19:27] Using ignore file (if present) at e:\PlusProgetti\Net.Pluservice.Motus\Unbounded\.prettierignore
["INFO" - 18:19:27] File Info:
{
  "ignored": false,
  "inferredParser": "typescript"
}
["INFO" - 18:19:27] Detected local configuration (i.e. .prettierrc or .editorconfig), VS Code configuration will not be used
["INFO" - 18:19:27] Prettier Options:
{
  "filepath": "e:\\PlusProgetti\\Net.Pluservice.Motus\\Unbounded\\src\\app\\views\\driverScheduleVersion\\driverScheduleVersion.component.ts",
  "parser": "typescript",
  "useTabs": false,
  "tabWidth": 2,
  "singleQuote": true,
  "printWidth": 120,
  "bracketSpacing": true,
  "trailingComma": "all"
}
["INFO" - 18:19:28] Formatting completed in 0.869ms.
```

It says there's a local .prettierrc and it's correct (code below), and it also reads `"printWidth": 120`, however, the line 128 chars long is not broken.

```
{
  "singleQuote": true,
  "printWidth": 120,
  "bracketSpacing": true,
  "trailingComma": "all"
}
```

If I run prettier from the cli, the output is correct. Should I look into the VSCode extension as the cli looks correct?

**Comment by duailibe** [comment]
@Balastrong Is it possible that the extension is using a different Prettier version than the one you're running on the CLI?

**Comment by Balastrong** [comment]
It is!

tl;dr Prettier is working, the issue was that two different versions were installed.

___
After spending much more time than I wanted to... turns out `package-lock.json` of the inner project had `2.2.1` while the outer project (the one that was taken for global formatting) `2.5.1`.

Making sure the same version was installed is something we did, but it didn't work so we gave up the last time we tried.
After closing VSCode and reopening a few hours later the magic happened and everything started working.

Just a guess, Prettier extension loads/caches Prettier when VSCode is launched and if you `npm install` other versions it keeps the previous one, that's what led us to the wrong path... but this is a whole another topic :D
Once again switching off and back on saved the situation!

**Comment by duailibe** [comment]
Glad to hear that it worked out!

As for breaking HTML classes, it's probably the same issue. We started doing that in 2.3.0 (#7865) and reverted it in 2.5.0 (#11827), so it's likely there are different versions in conflict.

**Comment by gkatsanos** [comment]
@duailibe any chance to have a `stable` branch that won't change formatting so radically ?

**Comment by duailibe** [comment]
@gkatsanos what do you mean by change formatting drastically?
